### PR TITLE
ci: added trivy scan gh action for docker images

### PR DESCRIPTION
**Pull request by rohit-bandlamudi-nr**
Trivy scan Github action to scan built image on every PR

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -35,9 +35,30 @@ jobs:
       - name: Build project
         run: make linux/amd64
 
+  # trivy-db-cache-init:
```
Comment (maya-jha) [inline .github/workflows/pr.yaml:38]: if this is not needed, remove.

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -93,6 +114,30 @@ jobs:
...
+          image-ref: registry:5000/fb-output-plugin-${{ matrix.name }}
+          format: table
+          ignore-unfixed: true
+          severity: 'MEDIUM,LOW'
+      
+      - name: Fail build on High/Criticial Vulnerabilities
+        env:
+          TRIVY_NON_SSL: true
+          TRIVY_DB_REPOSITORY: public.ecr.aws/aquasecurity/trivy-db
+        uses: aquasecurity/trivy-action@0.18.0
```
Comment (maya-jha) [inline .github/workflows/pr.yaml:133]: is this the latest: 0.18.0?

**Comment by maya-jha** [comment]
Why we are seeing so many vulnerabilities? Why the check is failing?

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -93,6 +114,30 @@ jobs:
...
           cache-from: type=local,src=/tmp/.buildx-cache
           cache-to: type=local,dest=/tmp/.buildx-cache
+          
+      - name: Run Trivy in table mode for (${{ matrix.name }})
+        env:
+          TRIVY_NON_SSL: true
+          TRIVY_DB_REPOSITORY: public.ecr.aws/aquasecurity/trivy-db
+        uses: aquasecurity/trivy-action@0.18.0
+        with:
+          image-ref: registry:5000/fb-output-plugin-${{ matrix.name }}
```
Comment (maya-jha) [inline .github/workflows/pr.yaml:124]: what does it refer to? Is it looking at debug image?

**Review by maya-jha — CHANGES_REQUESTED** [review]

**Comment by rohit-bandlamudi-nr** [comment]
> Why we are seeing so many vulnerabilities? Why the check is failing?

As discussed, we were wanting the workflow to fail if there are severe/high vulnerabilities.

File: .github/workflows/pr.yaml (hunk 2 above)
Comment (rohit-bandlamudi-nr) [reply .github/workflows/pr.yaml:133]: No. 0.28.0 is the latest version. I was referring to some repo that has this version. Will update it

File: .github/workflows/pr.yaml (hunk 3 above)
Comment (rohit-bandlamudi-nr) [reply .github/workflows/pr.yaml:124]: This action runs for all strategies in the matrix.

**Comment by rohit-bandlamudi-nr** [comment]
> Why we are seeing so many vulnerabilities? Why the check is failing?

As discussed, we were wanting the workflow to fail if there are severe/high vulnerabilities.

**Review by voorepreethi — DISMISSED** [review]

**Review by maya-jha — APPROVED** [review]

File: .github/workflows/pr.yaml [hunk 4]
```diff
@@ -93,6 +93,32 @@ jobs:
...
           cache-from: type=local,src=/tmp/.buildx-cache
           cache-to: type=local,dest=/tmp/.buildx-cache
+          
+      - name: Run Trivy in table mode for (${{ matrix.name }})
+        if: matrix.name != 'debug-image'
+        env:
+          TRIVY_NON_SSL: true
+          TRIVY_DB_REPOSITORY: public.ecr.aws/aquasecurity/trivy-db
+        # Versions 0.26.0 and above are causing an issue where the image can't be fetched because the image name is not being resolved.
+        uses: aquasecurity/trivy-action@0.25.0
```
Comment (voorepreethi) [inline .github/workflows/pr.yaml:103]: Can we point this to latest image

**Pull request merged**